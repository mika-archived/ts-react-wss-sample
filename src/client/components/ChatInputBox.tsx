import React, { useState } from "react";
import PropTypes from "prop-types";

import Button from "./controls/Button";
import TextBox from "./controls/TextBox";
import { HorizontalLayout } from "./controls/Layout";

type Props = {
  onSubmit: (message: string) => void;
};

type State = {
  message: string;
};

const initialState: State = {
  message: ""
};

const ChatInputBox: React.FC<Props> = props => {
  const [state, setState] = useState(initialState);
  const sendMessage = () => {
    props.onSubmit(state.message);
    setState({ message: "" });
  };

  return (
    <HorizontalLayout>
      <TextBox
        value={state.message}
        onChange={event => setState({ message: event.target.value })}
        style={{ flex: 1 }}
      />
      <Button primary disabled={!state.message} onClick={() => sendMessage()} type="submit" style={{ width: "100px" }}>
        送信
      </Button>
    </HorizontalLayout>
  );
};

ChatInputBox.propTypes = {
  onSubmit: PropTypes.func.isRequired
};

export default ChatInputBox;
