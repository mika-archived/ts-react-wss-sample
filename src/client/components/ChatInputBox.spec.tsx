import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

import React from "react";
import { cleanup, fireEvent, render } from "@testing-library/react";
import ChatInputBox from "./ChatInputBox";

afterEach(cleanup);

test("does not have changes between snapshots", () => {
  const { asFragment } = render(<ChatInputBox onSubmit={() => {}} />);

  expect(asFragment()).toMatchSnapshot();
});

test("fire `onSubmit` event with inputs and cleanup textbox when enter the message and click the button", () => {
  const callback = jest.fn();
  const { getByRole } = render(<ChatInputBox onSubmit={callback} />);

  fireEvent.change(getByRole("textbox"), { target: { value: "test" } });
  fireEvent.click(getByRole("button"));

  // fire event
  expect(callback).toHaveBeenCalledTimes(1);
  expect(callback).toHaveBeenCalledWith("test");

  // cleanup textbox
  expect((getByRole("textbox") as HTMLInputElement).value).toBe("");
});

test("if message is empty, can not click (fire) the submit button", () => {
  const callback = jest.fn();
  const { getByRole } = render(<ChatInputBox onSubmit={callback} />);

  fireEvent.click(getByRole("button"));

  expect(callback).toHaveBeenCalledTimes(0);
});
