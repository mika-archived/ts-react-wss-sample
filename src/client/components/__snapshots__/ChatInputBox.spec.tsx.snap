// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not have changes between snapshots 1`] = `
<DocumentFragment>
  .c2 {
  height: 1.8rem;
  padding: 0.25rem 0.6rem;
  color: #ffffff;
  background: #0074d9;
  border: 1px solid #0074d9;
}

.c2:disabled {
  color: #dddddd;
  background: #aaaaaa;
  border-color: #aaaaaa;
}

.c1 {
  height: 1.4rem;
  padding: 2px 8px;
  border: 1px solid #dddddd;
  outline: 0;
}

.c1:focus {
  border: 1px solid #0074d9;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
    class="c0"
    direction="horizontal"
  >
    <input
      class="c1"
      role="textbox"
      style="flex: 1;"
      value=""
    />
    <button
      class="c2"
      disabled=""
      role="button"
      style="width: 100px;"
      type="submit"
    >
      送信
    </button>
  </div>
</DocumentFragment>
`;
